### PR TITLE
Fix p3 publics

### PR DESCRIPTION
**Pull request by Schaeff**
p3 publics currently incorrectly use the column name as a key. This explains why this was working as long as all public inputs were the same column and value, even with different rows.

### CODE REVIEW

**Comment by leonardoalt** [comment]
I just tested with https://github.com/powdr-labs/powdr/pull/2046 and now I get this error:

```console
thread 'main' panicked at /home/leo/devel/powdr/plonky3/src/prover.rs:459:36:
no entry found for key
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

asm: https://gist.github.com/leonardoalt/7f59c09b7bc5229ef72e0416e348fda7

**Comment by Schaeff** [comment]
My bad, should have open draft, was still in progress. Thanks for testing though

**Comment by Schaeff** [comment]
> I just tested with #2046 and now I get this error:
> 
> ```
> thread 'main' panicked at /home/leo/devel/powdr/plonky3/src/prover.rs:459:36:
> no entry found for key
> note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
> ```
> 
> asm: https://gist.github.com/leonardoalt/7f59c09b7bc5229ef72e0416e348fda7

This is unrelated, will tackle in another PR.

File: plonky3/src/circuit_builder.rs [hunk 1]
```diff
@@ -418,13 +418,13 @@ where
...
         // public constraints
-        let public_vals_by_id = self
+        let public_vals_by_name = self
             .constraint_system
             .publics_by_stage
             .iter()
             .zip_eq(public_input_values_by_stage)
             .flat_map(|(publics, values)| publics.iter().zip_eq(values.iter()))
-            .map(|((id, _, _), pi)| (id, *pi))
+            .map(|((name, _, _, _), pi)| (name, *pi))
```
Comment (Schaeff) [inline plonky3/src/circuit_builder.rs:427]: This is where the issue was, we were addressing publics by the column name, not the public name

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**